packages/salesforcedx-vscode-core/src/commands/forceSourceDeployManifest.ts

```diff
@@ -52,49 +48,21 @@ export class ForceSourceDeployManifestExecutor extends BaseDeployExecutor {
   }
 }
 
-export class LibrarySourceDeployManifestExecutor extends LibraryCommandletExecutor<
-  string
-> {
+export class LibrarySourceDeployManifestExecutor extends DeployCommand<string> {
   constructor() {
     super(
-      'Deploy With Manifest (beta)',
-      'force_source_deploy_with_manifest',
-      OUTPUT_CHANNEL
+      nls.localize('force_source_deploy_text'),
+      'force_source_deploy_with_manifest_beta'
     );
   }
 
-  public async run(response: ContinueResponse<string>): Promise<boolean> {
+  protected async getComponents(
+    response: ContinueResponse<string>
+  ): Promise<ComponentSet> {
     const packageDirs = await SfdxPackageDirectories.getPackageDirectoryPaths();
-    try {
-      const components = await ComponentSet.fromManifestFile(response.data, {
-        resolve: packageDirs.map(dir => join(getRootWorkspacePath(), dir))
-      });
-      const deployPromise = components.deploy(
-        await workspaceContext.getConnection()
-      );
-      this.telemetry.addProperty(
-        'metadataCount',
-        JSON.stringify(createComponentCount(components))
-      );
-      const result = await deployPromise;
-
-      const outputResult = createDeployOutput(result, packageDirs);
-      channelService.appendLine(outputResult);
-      BaseDeployExecutor.errorCollection.clear();
-
-      if (result.status === DeployStatus.Succeeded) {
-        return true;
-      }
-
-      handleDeployRetrieveLibraryDiagnostics(
-        result,
-        BaseDeployExecutor.errorCollection
-      );
-
-      return false;
-    } finally {
-      await DeployQueue.get().unlock();
-    }
+    return ComponentSet.fromManifestFile(response.data, {
+      resolve: packageDirs.map(dir => join(getRootWorkspacePath(), dir))
+    });
```

Comment:
Now individual commands for the most part just have to extend the relevant base class (Deploy or Retrieve), and then describe how to fetch components.